react/src/tests/opReturn.test.ts

```diff
@@ -85,94 +107,109 @@ describe('getDataPushdata', () => {
+      allResultsPrefix +
+        '10' + // 16 in hex
+        '6d79437573746f6d5573657244617461',
+    );
+    expect(resultPaymentId?.length).toBe(paymentIdDigits);
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Simple parse OpReturn 1 no paymentId', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData', true) // 16 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-  })
+    const fullResult = encodeOpReturnProps('myCustomUserData', true); // 16 bytes
+    expect(fullResult).toBe(
+      allResultsPrefix +
+        '10' + // 16 in hex
+        '6d79437573746f6d5573657244617461' +
+        '00', // Empty paymentId pushdata
+    );
+  });
   it('Simple parse OpReturn 2', () => {
-    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data') // 33 bytes
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0,-paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix +
-      '21' + // 33 in hex
-      '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461')
-    expect(resultPaymentId?.length).toBe(paymentIdDigits)
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
+    const fullResult = encodeOpReturnProps('my=Longer more=sensible|user|data'); // 33 bytes
+    const resultPaymentId = fullResult?.slice(-paymentIdDigits);
+    const resultNoPaymentId = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultNoPaymentId).toBe(
+      allResultsPrefix +
+        '21' + // 33 in hex
+        '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461',
+    );
+    expect(resultPaymentId?.length).toBe(paymentIdDigits);
+    expect(resultPaymentId).toMatch(paymentIdRegex);
+  });
   it('Simple parse OpReturn 2 no paymentId', () => {
-    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data', true) // 33 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '21' + // 33 in hex
-      '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461')
-  })
+    const fullResult = encodeOpReturnProps(
+      'my=Longer more=sensible|user|data',
+      true,
+    ); // 33 bytes
+    expect(fullResult).toBe(
+      allResultsPrefix +
+        '21' + // 33 in hex
+        '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461' +
+        '00', // Empty paymentId pushdata
+    );
+  });
   it('Throws if too long', () => {
-    const data208bytes = '😂'.repeat(52) // 52 * 4 = 208 bytes
-    expect(() =>
-      parseOpReturnProps(data208bytes)
-    ).toThrow('Maximum 205 byte size exceeded for user data: 208')
-  })
+    const data208bytes = '😂'.repeat(52); // 52 * 4 = 208 bytes
+    expect(() => encodeOpReturnProps(data208bytes)).toThrow(
+      'Maximum 205 byte size exceeded for user data: 208',
+    );
+  });
   it('Is not too long if no paymentId', () => {
```

Comment:
`it('Supports data push exceeding 205 bytes if no nonce')`

need to add these tests

1. No nonce, max data push
2. No nonce, data push still too big

Reply:
added.